pkg/cases/handler.go

```diff
@@ -8,14 +8,146 @@ import (
 // @Description
+// @Tags Case
 // @Produce json
-// @Param  id path uint true "case ID"
+// @Param patientID path uint true "patient ID"
 // @Success 200 {object} api.ReturnedData{data=Case}
-// @Router /case/{id} [GET]
-func (h *CaseHandler) GetCaseByCaseID(c echo.Context) error {
-	// what is in here does not matter for now, just filling the swaggo comments and name the function
-	c.Logger().Debug("hello world")
-	return c.JSON(200, api.Return("pong", nil))
+// @Router /patient/{patientID}/case [GET]
+func (h *CaseHandler) GetLastCase(c echo.Context) error {
+	// ...
+	c.Logger().Debug("GetLastCase")
+	return c.JSON(200, api.Return("success", nil))
+}
+
+// @Summary Get cases by patient ID
+// @Description
+// @Tags Case
+// @Produce json
+// @Param patientID path uint true "patient ID"
+// @Param department query string false "department name" nil
+// @Success 200 {array} api.ReturnedData{data=[]uint}
+// @Router /patient/{patientID} [GET]
+func (h *CaseHandler) GetCasesByPatientID(c echo.Context) error {
+	// ...
+	c.Logger().Debug("GetCasesByPatientID")
+	return c.JSON(200, api.Return("success", nil))
+}
+
+// @Summary New a case
+// @Description
+// @Tags Case
+// @Produce json
+// @Param caseDetail body Case true "patient ID, doctor ID, department name and other case details"
+// @Success 200 {object} api.ReturnedData{data=Case}
+// @Router /patient/{patientID}/case [POST]
+func (h *CaseHandler) NewCase(c echo.Context) error {
+	// ...
+	c.Logger().Debug("NewCase")
+	return c.JSON(200, api.Return("success", nil))
+}
+
+// @Summary Delete a case
+// @Description
+// @Tags Case
+// @Produce json
+// @Param caseID path uint true "case ID"
+// @Success 200 {object} api.ReturnedData{}
+// @Router /patient/{patientID}/case/{caseID} [DELETE]
+func (h *CaseHandler) DeleteCaseByCaseID(c echo.Context) error {
+	// ...
+	c.Logger().Debug("DeleteCaseByCaseID")
+	return c.JSON(200, api.Return("success", nil))
+}
+
+// @Summary Get prevoius cases
+// @Description
+// @Tags Case
+// @Produce json
+// @Param caseID path uint true "current case ID"
+// @Success 200 {object} api.ReturnedData{data=[]Case}
+// @Router /patient/{patientID}/case/{caseID} [GET]
+func (h *CaseHandler) GetPreviousCases(c echo.Context) error {
+	// ...
+	c.Logger().Debug("GetPreviousCases")
+	return c.JSON(200, api.Return("success", nil))
+}
+
+// @Summary Update case
+// @Description
+// @Tags Case
+// @Produce json
+// @Param caseDetail body string true "case ID and updated details"
+// @Success 200 {object} api.ReturnedData{data=Case}
+// @Router /patient/{patientID}/case/{caseID} [PUT]
+func (h *CaseHandler) UpdateCase(c echo.Context) error {
+	// ...
+	c.Logger().Debug("UpdateCase")
+	return c.JSON(200, api.Return("success", nil))
+}
+
+// @Summary New a prescrition
+// @Description
+// @Tags Case
+// @Produce json
+// @Param prescriptionDetail body string true "case ID and prescription details"
+// @Success 200 {object} api.ReturnedData{data=Prescription}
```

Comment:
```suggestion
// @Success 200 {object} api.ReturnedData{data=nil}
```
Why do you have to return a data after creation?